tensorboard/webapp/metrics/store/metrics_reducers_test.ts

```diff
@@ -2484,6 +2494,77 @@ describe('metrics reducers', () => {
 
         expect(nextState1.useRangeSelectTime).toEqual(false);
       });
+
+      it('sets `cardStepIndexMap` on selected time changed', () => {
+        const beforeState = buildMetricsState({
+          selectTimeEnabled: false,
+          cardMetadataMap,
+          timeSeriesData: {
+            ...buildTimeSeriesData(),
+            images: {
+              tagC: {
+                111: {
+                  runToLoadState: {},
+                  runToSeries: {
+                    'test run Id': [
+                      {step: 10, wallTime: 0, imageId: '1'},
+                      {step: 20, wallTime: 10, imageId: '2'},
+                      {step: 30, wallTime: 15, imageId: '3'},
+                    ],
+                  },
+                },
+              },
+            },
+          },
+          cardStepIndex: {[imageCardId]: 2},
+        });
+
+        const nextState = reducers(
+          beforeState,
+          actions.timeSelectionChanged({
+            startStep: 10,
+            endStep: undefined,
+          })
+        );
+
+        expect(nextState.cardStepIndex).toEqual({[imageCardId]: 0});
+      });
+
+      it('does not set `cardStepIndexMap` on selected time changed', () => {
```

Comment:
cardStepIndex? Also, state why this one does not set cardStepIndex while above test does set it.

Reply:
Good point. Updated the description to make these two tests meaningful on why are they different.
also remove "on selected time changed" since all tests are under "#timeSelectionChanged" already